src/components/ItemsList.js

```diff
@@ -1,38 +1,72 @@
 import React from 'react';
 import firebase from '../lib/firebase';
 import Nav from './Nav';
+import '../styles/ItemsList.css';
 import { useCollectionData } from 'react-firebase-hooks/firestore';
 
 const db = firebase.firestore().collection('shopping_list');
 
+const wasItemPurchasedWithinLastOneDay = (lastPurchasedOn) => {
+  if (lastPurchasedOn === null) return false;
+  const oneDayInMilliseconds = 24 * 60 * 60 * 1000;
+  return Date.now() - lastPurchasedOn <= oneDayInMilliseconds;
+};
+
 const ItemsList = () => {
   const userToken = localStorage.getItem('token');
 
   const [shoppingList, loading, error] = useCollectionData(
     db.where('token', '==', userToken),
+    { idField: 'documentId' },
   );
 
+  const markItemAsPurchased = (index) => {
+    const { items, documentId } = shoppingList[0];
+
+    if (items[index].lastPurchasedOn === null) {
+      items[index].lastPurchasedOn = Date.now();
+    } else {
+      items[index].lastPurchasedOn = null;
+    }
```

Comment:
This can be shortened to one liner with a ternary operator 
`items[index].lastPurchasedOn = items[index].lastPurchasedOn ? null : Date.now();`